Comments for changes

oj-kentd opened the pull request.

BryceVichiOJ commented:
Makes sense to me. I'll make these changes and update this PR.

BryceVichiOJ commented:
All requested changes were put in place. Clear to begin review process.

nichmj01 commented on include/parakeet/macros.h:
```diff
@@ -0,0 +1,9 @@
+*/
+
+#if defined(_WIN32)
+    #define DEPRECATED(func) __declspec(deprecated) func
```
We probably want to namespace this macro to avoid potential conflicts; we've run into that problem with the openjaus code base. Something like PARAKEET_DEPRECATED or PRKT_DEPRECATED if we're okay with using PRKT as an abbreviation.

- [x] Rename DEPRECATED macro

BryceVichiOJ replied:
Went with PARAKEET_DEPRECATED.

nichmj01 commented on include/parakeet/util.h:
```diff
@@ -9,13 +9,10 @@
+#if defined(_WIN32)
     #define M_PI   3.14159265358979323846264338327950288
 #endif
 
```
Maybe we should put a check to ensure M_PI is defined and generate #error if not? Not sure if the compile will complain later or not?

BryceVichiOJ replied:
By that logic, we could simply check if M_PI is already defined, and only override it if not.

nichmj01 replied:
Agreed that's a better solution.

- [x] Only #define M_PI if it's not already defined.

BryceVichiOJ replied:
Done

nichmj01 commented on src/parakeet/Driver.cpp:
```diff
@@ -56,6 +56,11 @@ namespace parakeet
             return false;
         }
 
+        g_withIntensity = intensity;
```
Not a fan of the use 'g_' at the start of the variable names here. Also, in light of the book club conversations, it might be worthwhile to clean up this code a little as it's the most complex and most important one in this project.

The \brief for the connect method in the header is also wrong.

BryceVichiOJ replied:
Good catch on the \brief.

There are several things I'd like to do to the Driver class based off the bookclub (having a struct for parameter is a big one). Should the scope of this PR to match all those changes?

nichmj01 replied:
Hmmm... you being up a good point. Let's keep these changes scoped to just addressing the units issue and add a YT issue for general clean up which we can work on soon.

BryceVichiOJ replied:
I modified the \brief to be correct, created YT Issue PRKT-82 to handle the rest of the comments.

oj-kentd reviewed (commented):
My concerns with units are addressed.

BryceVichiOJ committed:
Add Units to Variable and Function Names

- Added units to certain variables and functions
- Standardized our linux/windows if defines
- Added deprecation macro
